src/main/scala/com/codacy/client/bitbucket/client/BitbucketClientBase.scala

```diff
@@ -243,15 +241,15 @@ abstract class BitbucketClientBase(credentials: Credentials) {
   }
 
   private def withClientEither[T](block: WSClient => Either[ResponseError, T]): Either[ResponseError, T] = {
-    withClient(block) match {
+    Try(block(client)) match {
```

Comment:
Codacy found an issue: [Pattern matching is not exhaustive.](https://app.codacy.com/gh/codacy/bitbucket-scala-client/pullRequest?prid=5921392)